File: javascript/fat.js
function fat(x) {
	function fat_aux(x1, acum) {
		// if (x1 === 0)
		if (x1 === 0) // Sugestão do https://repl.it para ===
			return acum;
		else
			return fat_aux(x1 - 1, acum*x1);
	}
	return fat_aux(x, 1);
}

File: C#/fat.cs
void fat (int x, out int ac) {
	if (x > 0) {
		ac = ac * x;
		fat (x-1, out ac);
	}
}

int fat2 (int x) {
	if (x > 0) 
		return x * fat2(x-1);
	else
		return 1;
}

